src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -447,6 +462,13 @@ public Set<ItemStorage> getCopyOfPlantables()
         return new HashSet<>(plantables);
     }
 
+    @Override
+    public Set<ItemStorage> getCopyOfFlowers()
+    {
+        if (flowers.isEmpty()) Log.getLogger().error("getCopyOfFlowers when empty");
+        return new HashSet<>(flowers);
```

Comment:
Use immutable collection instead of returning a copy

Reply:
Should be a part of a bigger rework. We partially use the copies filter them down etc, so would have to be very careful. The methods are called "getCopy" on purpose atm.

Reply:
don't think we should keep repeating bad behaviour though :D the bigger rework is unlikely to ever happen